### DISCUSSION TITLE
MassUpload is failing

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#283] [Q&A] [answered]
I'm trying to upload a bunch of 3-4 MB PNG files via the Mass Uploader. 
Often, as soon as the upload starts, upload fails, so it looks like this:

![image](https://user-images.githubusercontent.com/1337223/152598684-efea7856-b274-4a1e-9076-f65d61481c69.png)

in the logs I find this.

```
[2022-02-04 20:24:56] Log.ERROR: API [POST]:/api/v1/:controller/action/:action, Params:Array (     [controller] => Attachment     [action] => CreateByChunks ) , InputData: {"chunkId":"49ae0e6d4de04791ac975bbf210fbc03","name":"FILE.png","type":"image/png","size":3491777,"role":"Attachment","relatedType":"Asset","field":"files","modelAttributes":{"libraryHasDefaultAttributes":true,"afterSaveMessage":"","isActive":true,"libraryId":"1","libraryName":"Default Library","private":false,"type":"Gallery Image","name":"massUpload","filesIds":[],"filesNames":{},"filesTypes":{},"ownerUserName":"Admin","ownerUserId":"1","assignedUserName":"Admin","assignedUserId":"1"}} -  [] []
[2022-02-04 20:24:56] Log.ERROR: Display Error: , Code: 500 URL: /api/v1/Attachment/action/CreateByChunks?silent=true [] []
```

I can retry the upload. Sometimes it works or it says Mimetype is not supported.

![image](https://user-images.githubusercontent.com/1337223/152598875-670ac192-f963-4d31-a898-aa53ff6ad547.png)

In the settings I set no duplicate files - check by filename (I remember I had issues if it checks the file content for duplicates)

I believe the wrong Mime Type message stems from existing files that got copied from chunks/ to the file storage, but got damaged in the process, by something. 

So this is kind of broken :-( 

Is there a way to debug this and fix it? What to look for?

Thanks!

**Comment by marcus-at-localhost** [comment]
I just changed the key `attachmentUploadMaxSize` in the config that I found here: https://github.com/atrocore/atrocore/blob/dcb7a2aab8e2b774ad7ce1cabacf66b90d5d659b/app/Espo/Services/App.php#L115 - now it seems to work?! 

Is that a user setting in the backend? I feel I missed something

**Comment by atrolex** [comment]
There are some settings in the config files, which are not available in the Administration area. We believe these should be touched in very rare cases.  

I think we have maximum chunk size as a config parameter. "attachmentUploadMaxSize" could be max file size which is uploaded NOT in chunks. I am not sure ..

1) Are you sure incorrect MIME type is "false positive"? 
2) Do you have every time different files marked as "incorrect MIME type" or always the same?
3) Are you sure your server has no issues? Have you tried to upload the same bunch of assets on another server or locally? You could also test it on demo.atropim.com and than delete your assets from there..
4) How many files are you trying to mass upload at once? Not too much, right?

**Reply by marcus-at-localhost** [reply]
> There are some settings in the config files, which are not available in the Administration area. We believe these should be touched in very rare cases.
> 
> I think we have maximum chunk size as a config parameter. "attachmentUploadMaxSize" could be max file size which is uploaded NOT in chunks. I am not sure ..

[When I checked the code, that negotiates the max filesize](https://github.com/atrocore/atrocore/blob/dcb7a2aab8e2b774ad7ce1cabacf66b90d5d659b/app/Espo/Services/App.php#L115) I noticed the `attachmentUploadMaxSize` config key (and it reads from the php.ini) and checked my config. There I only saw `inlineAttachmentUploadMaxSize => 20` - not sure where this mistake came from, but I corrected this and set it to a higher number (attachmentUploadMaxSize = 20000) and now it works.

/data/chunks/ are still being created (2MB chunks it seems)

> 
> 1. Are you sure incorrect MIME type is "false positive"?

Yes the type was correct for the original file

> 2. Do you have every time different files marked as "incorrect MIME type" or always the same?

I think once it got to the incorrect MIME type I could not reset it. I checked the target folder where the file is moved to after uploading the chunks and there was an incomplete file in there. So I believe while creating the file from chunks the process might been interrupted and the system recognizes a file, but it's broken.

> 3. Are you sure your server has no issues? Have you tried to upload the same bunch of assets on another server or locally? You could also test it on demo.atropim.com and than delete your assets from there..

So far everything worked well, so I assume the Server Setup is alright, and I believe with my change in config I solved it.

> 4. How many files are you trying to mass upload at once? Not too much, right?

Yeah, 5 to 8 files a ~3MB at once. Nothing too crazy :)

What is `'fileUploadStreamCount' => 2,`? Parallel upload? I set that to 2 I believe. Maybe that helped too?

So, I'll keep an eye on that and report if I notice something else.
Thanks for getting back though.

**Comment by atrolex** [comment]
I think we have found it, thx @marcus-at-localhost :) It has to do with a low amount of chunks (only 2 in your case). System tries to combine them, despite not all are already uploaded. I think this problem will be fixed with the next release of core.

**Comment by atrolex** [comment, marked as answer]
@marcus-at-localhost we have done some refactoring in core and dam, please update and test again.
